crates/eips/src/eip7702/auth_list.rs

```diff
@@ -0,0 +1,140 @@
+use core::ops::Deref;
+
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+use alloy_primitives::{keccak256, Address, ChainId, B256};
+#[cfg(feature = "k256")]
+use alloy_primitives::{Signature, SignatureError};
+use alloy_rlp::{BufMut, Decodable, Encodable, Header, RlpDecodable, RlpEncodable};
+
+/// An unsigned EIP-7702 authorization.
+#[derive(Debug, Clone, RlpEncodable, RlpDecodable)]
+pub struct Authorization {
+    chain_id: ChainId,
+    address: Address,
+    nonce: OptionalNonce,
```

Comment:
can we make these pub, otherwise there's no way to instantiate them

Reply:
I'd prefer a constructor